lib/commands/element.js

```diff
@@ -107,13 +132,9 @@ commands.setValue = async function (value, el) {
     await this.executeAtom('click', [atomsElement]);
     await this.executeAtom('type', [atomsElement, value]);
   } else {
-    // first make sure that the keyboard is up
-    try {
-      await this.findNativeElementOrElements('class name', 'XCUIElementTypeKeyboard', false);
-    } catch (err) {
-      // no keyboard found
-      log.debug('No keyboard found. Clicking element to open it.');
-      await this.nativeClick(el);
+    // make sure there is a keyboard if this is a text field
+    if (await this.getAttribute('type', el) === 'XCUIElementTypeTextField') {
```

Comment:
yes, this seems like a very important check. don't want us randomly clicking on other things :-)